### PR TITLE
#4383 update react native

### PR DESCRIPTION
**Pull request by desfero**
**Technical notes**
This pr updates react-native to the latest stable version.

Blog posts
https://reactnative.dev/blog/2020/03/26/version-0.62
https://reactnative.dev/blog/2020/07/06/version-0.63

the most interesting changes are Flipper suppor, improved debugging experience with logbox and Pressable component.

**TODO**
- [x] Make sure flipper is working
- [x] Fix deprecation warnings if possible

Follow up ticket https://github.com/Neufund/platform-frontend/issues/4513

### CODE REVIEW

File: packages/wallet/app/components/screens/WalletScreen/Transactions/Transactions.tsx [hunk 1]
```diff
@@ -13,7 +13,8 @@ import { spacingStyles } from "styles/spacings";
 import { ItemSeparatorComponent } from "./ItemSeparatorComponent";
 import { getTxUniqueId } from "./utils";
 
-type TFlatList = React.ComponentProps<typeof Animated.FlatList>;
+declare class TypedFlatList extends Animated.FlatList<TTransaction> {}
+type TFlatList = React.ComponentProps<typeof TypedFlatList>;
```
Comment (desfero) [inline packages/wallet/app/components/screens/WalletScreen/Transactions/Transactions.tsx:17]: i don't really know how to type that properly with generic data provided  `React.ComponentProps<TypedFlatList<TTransaction>>` is not working and this is what i come up with 
```
declare class TypedFlatList extends Animated.FlatList<TTransaction> {}
type TFlatList = React.ComponentProps<typeof TypedFlatList>;
```

File: packages/wallet/ios/wallet-Bridging-Header.h [hunk 2]
```diff
@@ -0,0 +1,4 @@
+//
```
Comment (desfero) [inline packages/wallet/ios/wallet-Bridging-Header.h:1]: this is required for xcode to include `swift` binaries

File: packages/wallet/app/components/screens/AccountBackupScreen/AccountBackupScreenLayout.tsx [hunk 3]
```diff
@@ -31,6 +31,7 @@ const Data: React.FunctionComponent<TExternalProps> = ({ data, allowedTypes, onT
     animationRef.current.setValue(0);
 
     Animated.timing(animationRef.current, {
+      useNativeDriver: false,
```
Comment (arjunu) [inline packages/wallet/app/components/screens/AccountBackupScreen/AccountBackupScreenLayout.tsx:34]: why `false`?

File: packages/wallet/app/store/create.tsx [hunk 4]
```diff
@@ -1,12 +1,25 @@
...
 
 import { setupInitModule } from "modules/init/module";
 
 import { TAppGlobalState } from "./types";
 
+export function getFlipperIntegrationExtension() {
+  // eslint-disable-next-line @typescript-eslint/no-require-imports, @typescript-eslint/no-var-requires, @typescript-eslint/no-unsafe-assignment
+  const { default: createDebugger } = require("redux-flipper");
+
+  invariant(isFunction(createDebugger), "createDebugger is not a function");
```
Comment (arjunu) [inline packages/wallet/app/store/create.tsx:16]: why do we need to check this?

File: packages/wallet/app/components/screens/AccountBackupScreen/AccountBackupScreenLayout.tsx (hunk 3 above)
Comment (desfero) [reply packages/wallet/app/components/screens/AccountBackupScreen/AccountBackupScreenLayout.tsx:34]: it's a default value and now it's a required key so the transition to `true` as a default one is easier. Here we animate something that's not supported by native driver AFAIK

File: packages/wallet/app/store/create.tsx (hunk 4 above)
Comment (desfero) [reply packages/wallet/app/store/create.tsx:16]: just to typecast so there is no need to force the cast to a function

**Pull request merged**